rs/ethereum/ledger-suite-orchestrator/test_utils/BUILD.bazel

```diff
@@ -3,12 +3,11 @@ load("@rules_rust//rust:defs.bzl", "rust_library")
 package(default_visibility = ["//visibility:public"])
 
 rust_library(
-    name = "test_utils",
```

Comment:
@basvandijk : I'm a bit puzzled by the motivation 

> it's hard to know which test_utils target is being referenced making it harder to know which crate to optimise.

Since it feels to me, but admittedly I don't have the full context here, we are optimizing for one particular tool at the disadvantage of developers: for example, if I develop something in the crate with the _path_ `/rs/ethereum/ledger-suite-orchestrator/test_utils` and I want to check that it still builds, I can just do `bazel build //rs/ethereum/ledger-suite-orchestrator/test_utils` _without looking_ at the Bazel file to build that particular crate. With your suggested change, I would need to parse the correct Bazel file to find out the exact name of the target, which contains a lot of redundant information with respect to the path (`ic_ledger_suite_orchestrator_test_utils` is redundant because I already know we are in `/rs/ethereum/ledger-suite-orchestrator/` in the IC repository).

I understand that another use-case is to have statistics across the whole repository, but shouldn't the tool for such jobs "just" display the full path of the target? Happy to discuss offline if that helps.

Reply:
I was kind of expecting this response and it's also the gripe I have with my own proposal. A nicer solution would indeed be if bazel would store the full target path in its profile instead of just the last component. We could look into patching bazel for that but it will likely take quite some time until we can use that modified version (if the bazel developers even accept our patch).

What about a compromise where, in addition to the renames in this PR, we rename the crate's directory to the crate's name? As in:
`/rs/ethereum/ledger-suite-orchestrator/test_utils` ->
`/rs/ethereum/ledger-suite-orchestrator/ic_ledger_suite_orchestrator_test_utils`

That way you can still build that crate using `//rs/ethereum/ledger-suite-orchestrator/ic_ledger_suite_orchestrator_test_utils` without looking at the bazel file.

Reply:
Thanks @basvandijk for your explanation!

> A nicer solution would indeed be if bazel would store the full target path in its profile instead of just the last component.

I think this was the key piece I was missing 💡 . From reading the PR description, I thought the problem was by this `superset` tool and not by Bazel, which is indeed a very different problem (maybe worth mentioning in the PR description?)

> What about a compromise where, in addition to the renames in this PR, we rename the crate's directory to the crate's name? 

Thanks for suggesting a compromise! I think for now I would not do it (renaming the crate's directory) and only do it if there are too many complaints (one the plus side it does solve the aforementioned problem, but on the downside  every such dir would contain a somewhat long "zero-knowledge" prefix that would make it more cumbersome to read in a tree in a GUI for example). For now, one could also re-build everything in a given directory (`/rs/ethereum/ledger-suite-orchestrator/test_utils/...`) and let Bazel caching do its magic)

Reply:
@basvandijk  do you know how critical path statistics would look like if we do the following:
* rename the target (as we do now)
* add a `test_utils` alias to the new target
* reference the alias instead of the renamed target

For example
```bazel
alias(
    name = "test_utils",
    actual = ":ic_crypto_internal_threshold_sig_canister_threshold_sig_test_utils",
)
```

Will we see the alias in the statistics or the full target name?

Reply:
Thanks for the approve. Let's see if more folks complain about this and if so I'll reconsider.

In the mean time I'll look into whether the Bazel Event Protocol somehow supports retrieving the full target paths.

Reply:
> Will we see the alias in the statistics or the full target name?

Interesting question. I don't know but I could find out.

Reply:
I agree with @gregorydemay here: ideally this change was not necessary. Given how useful the change is for the critical path statistics, the change also seems fine to me, however.

If Bazel would support retrieving the full path, that would be ideal I guess. Patching Bazel might also be an option, but that would clearly take some time.